### PR TITLE
Add implicit (delegated) partner to Applicants

### PR DESCRIPTION
**Pull request by dreispt**

### CODE REVIEW

File: hr_recruitment_partner/README.rst [hunk 1]
```diff
@@ -0,0 +1,79 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
+Applicants implicit Partner
+===========================
+
+Automatically create a Partner for Applicants.o
+
```
Comment (saltonmassally) [inline hr_recruitment_partner/README.rst:8]: .o at the end of the line is a typo i think

File: hr_recruitment_partner/init_hooks.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+def post_init_hook(cr, pool):
+    env = Environment(cr, SUPERUSER_ID, {})
+    # create a new partner for all applicants with no real partner assigned
+    applicants = env['hr.applicant'].with_context(active_test=False).\
+        search([('partner_id', '=', 1)], order='id')
+    # we need to read related values from the database because the related
+    # fields already cover our fields
+    if not applicants.ids:
+        return
```
Comment (saltonmassally) [inline hr_recruitment_partner/init_hooks.py:39]: Is "if not applicants" not enough to achieve this?

File: hr_recruitment_partner/init_hooks.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    env = Environment(cr, SUPERUSER_ID, {})
+    env.cr.execute(
+        'UPDATE hr_applicant SET partner_id=1 WHERE partner_id IS NULL')
+
+
+def post_init_hook(cr, pool):
+    env = Environment(cr, SUPERUSER_ID, {})
+    # create a new partner for all applicants with no real partner assigned
+    applicants = env['hr.applicant'].with_context(active_test=False).\
+        search([('partner_id', '=', 1)], order='id')
```
Comment (saltonmassally) [inline hr_recruitment_partner/init_hooks.py:35]: Is there a need for "order='id'"?

**Comment by saltonmassally** [comment]
tiny non blocking issue :+1:

File: hr_recruitment_partner/init_hooks.py (hunk 3 above)
Comment (dreispt) [inline hr_recruitment_partner/init_hooks.py:35]: It's redundant now, but it's there just in case some other module changes the model's order.

**Comment by dreispt** [comment]
Fixed.

File: hr_recruitment_partner/README.rst [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+for existing Applicants not having one.
+
+
+Configuration
+=============
+
+On install, new Partner record are automatically created for existing
+Applicants without one.
+
+You may like to custiomize your Applicants form view to add it any Partner
```
Comment (fjouatte) [inline hr_recruitment_partner/README.rst]: minor typo : "custiomize" -> "customize"

File: hr_recruitment_partner/README.rst [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+address details.
+
+When installing the first time, related Partners will automatically be created
+for existing Applicants not having one.
+
+
+Configuration
+=============
+
+On install, new Partner record are automatically created for existing
```
Comment (fjouatte) [inline hr_recruitment_partner/README.rst]: This sentence seems to be redundant. You just wrote the same thing few lines upper.

**Comment by fjouatte** [comment]
It needs unit tests isn't it ? I'll do it if you're ok with that

**Comment by fjouatte** [comment]
Maybe I'm wrong and  I don't understand the module but the readme says that it automatically creates the partner when creating a new application. That's not the case, right ? It just adds the constraint to say that the partner_id is required and create a partner for the already created application thanks to the pre and post hooks ? 

Sorry if I'm wrong

**Comment by dreispt** [comment]
@fjouatte If you could help with the unit test, I would appreciate it very much.

> Maybe I'm wrong and I don't understand the module but the readme says that it automatically creates the partner when creating a new application. That's not the case, right ? It just adds the constraint to say that the partner_id is required and create a partner for the already created application ? 

It does, you can try it. The mechanism doing it is the delegation inheritance: `delegate=True`

**Comment by dreispt** [comment]
Fixed.

**Comment by dreispt** [comment]
On tests, the only code file in the module already had [91% coverage](https://coveralls.io/files/1351968335). No additional logic other than that, so probably won't add much.

Any other action needed?

**Comment by fjouatte** [comment]
@dreispt you're right, tests are ok. Does post_init_hook need the pool argument ?

**Comment by dreispt** [comment]
@fjouatte Yes, the function's signature requires those two positional arguments.

**Comment by dreispt** [comment]
Needs reviews please.

**Comment by feketemihai** [comment]
:+1: 
Tested on runbot also and it's working ok.

**Comment by dreispt** [comment]
Fixed the README.
All good to merge?

File: hr_recruitment_partner/README.rst [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+
+Usage
+=====
+
+Some fields in the Applicant form are now mapped to the corresponding fields
+in the partner form. For example, editing the Phone or Email in one place will
+implicitly change it in the other.
+
+
```
Comment (feketemihai) [inline hr_recruitment_partner/README.rst:38]: @dreispt Could you add the Try on Runbot section...please

**Comment by dreispt** [comment]
Done

**Comment by feketemihai** [comment]
:+1:

**Comment by dreispt** [comment]
Please merge.

**Comment by ghost** [comment]
:+1:

**Comment by dreispt** [comment]
Thanks!

**Pull request merged**